contracts/oracles/OracleManager.sol

```diff
@@ -131,22 +151,22 @@ contract OracleManager is IOracleManager {
       // Get Previous round data to validate correctness.
       (, , uint256 previousOracleUpdateTimestamp, , ) = chainlinkOracle.getRoundData(oracleRoundIdsToExecute[i] - 1);
 
-      // Check if the previous oracle timestamp was zero, but the current one wasn't - then check if there was a phase change.
-      if (previousOracleUpdateTimestamp == 0 && (oracleRoundIdsToExecute[i] >> 64 != (oracleRoundIdsToExecute[i] - 1) >> 64)) {
-        uint16 numberOfPhaseChanges = 1;
-
+      // Breakdown of the below check:
+      //  - the previous oracle timestamp was zero
+      //  &&
+      //  - the previous price was from a previous phase. The previous price is from a previous phase if the aggregatorRoundId is 1 (since it is the first round)
+      if (previousOracleUpdateTimestamp == 0 && ((oracleRoundIdsToExecute[i] << 16) >> 16) == 1) {
         // NOTE: if the phase changes, then we want to correct the phase of the update.
         //       There is no guarantee that the phaseID won't increase multiple times in a short period of time (hence the while loop).
         //       But chainlink does promise that it will be sequential.
         // View how phase changes happen here: https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/dev/AggregatorProxy.sol#L335
-        while (previousOracleUpdateTimestamp == 0) {
-          uint16 prevPhaseId = uint16((oracleRoundIdsToExecute[i] >> 64) - numberOfPhaseChanges++);
-          if (lastRoundId[prevPhaseId] != 0) {
-            // NOTE: re-using this variable to keep gas costs low for this edge case.
-            (, , previousOracleUpdateTimestamp, , ) = chainlinkOracle.getRoundData(lastRoundId[prevPhaseId]);
-          } else {
-            revert NoPhaseIdSet({phaseId: prevPhaseId});
-          }
+        uint16 newPhaseId = uint16(oracleRoundIdsToExecute[i] >> 64);
+        if (roundIdOfLastPriceBeforePhaseStart[newPhaseId] != 0) {
+          // NOTE: re-using this variable to keep gas costs low for this edge case.
+          (, , previousOracleUpdateTimestamp, , ) = chainlinkOracle.getRoundData(roundIdOfLastPriceBeforePhaseStart[newPhaseId]);
```

Comment:
We use the value that gets set from the `setRoundIdOfLastPriceBeforePhaseStart` function.